vllm/v1/core/guided_decoding/grammar.py

```diff
@@ -0,0 +1,154 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import (TYPE_CHECKING, Any, Generic, Literal, TypeVar, get_args,
+                    overload)
+
+from typing_extensions import Annotated, LiteralString
+
+from vllm.utils import LazyLoader
+
+if TYPE_CHECKING:
+    import torch
+    import xgrammar as xgr
+    from typing_extensions import Self
+else:
+    xgr = LazyLoader("xgr", globals(), "xgrammar")
+    torch = LazyLoader("torch", globals(), "torch")
```

Comment:
Do we get benefit from lazy-loading torch? Is there ever a case when vllm would not import torch?